### PR TITLE
TBOX-223 Golden Records Delta pipeline example

### PR DESCRIPTION
**Pull request by ycharm**
# ↪️ Pull Request

## ✔️ PR Todo

- [x] Add documentation
- [x] Add examples
- [x] Ensure you are following the practices in the [contributor guide](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit) and [coding_standards](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards)
  * [Use consistent and descriptive name conventions](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.477s36w4rds9) 
  * [Uses google-style docstrings](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Google-style-docstrings)
  * [Uses versioned API/client methods only](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.yv6df99fyrq2)
  * [Avoid global variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-global-variables)
  * [Avoid giant try/catch](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-giant-try-/-except)
  * [Don't use mutable default variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Don%E2%80%99t-use-mutable-default-arguments)
  * [Use a `main()` function in your scripts](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Use-a-%E2%80%9Cmain()%E2%80%9D--function-in-your-scripts)
- [x] Added/updated testing for this change (ensure you are testing any changes/new code!)
- [x] Update title to link to TBOX issue (i.e. start pull request title with TBOX-xx) 
- [ ] Approval from first code-reviewer
- [ ] Approval from second code-reviewer (only required if a second reviewer has left comments)
- [x] Pass all checks

### CODE REVIEW

File: examples/scripts/project/golden_records/run_golden_records_delta.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
+"""Script for running a Mastering Pipeline upto the point of Updating Golden Records"""
+# %%
```
Comment (skalish) [inline examples/scripts/project/golden_records/run_golden_records_delta.py]:
```suggestion
```
Should this be here?  I wonder if this is confusing the doc build.

File: examples/scripts/project/golden_records/run_golden_records_delta.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        tamr_client.projects.by_resource_id(GR_project_id)
+    )
+
+    if upd_gr_sts[0].state == "SUCCEEDED":
+        PrjRunStatus["update_GR_Dataset"] = "Successful"
+        LOGGER.info("Golden Records Updated")
+    else:
+        LOGGER.info("FAILURE to update Golden Records")
+
+    LOGGER.info("Advaning pinned version of CHANGES dataset")
```
Comment (Kovacsova) [inline examples/scripts/project/golden_records/run_golden_records_delta.py]: typo

File: examples/scripts/project/golden_records/run_golden_records_delta.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    PrjRunStatus["update_GR_MapDeltas_project"] = "Failed"
+    PrjRunStatus["publish_GR"] = "Failed"
+    PrjRunStatus["update_GR_output"] = "Failed"
+    PrjRunStatus["update_GR_Delta"] = "Failed"
+    PrjRunStatus["update_GR_Mapping"] = "Failed"
+
+    # Create the tamr client
+    tamr_client = utils.client.create(**instance_connection_info)
+    LOGGER.info(
+        "Retreiving project names and resource IDs from Tamr and storing as a lookup dictionary"
```
Comment (Kovacsova) [inline examples/scripts/project/golden_records/run_golden_records_delta.py]: typo

**Comment by skalish** [comment]
Closing for now as we need to wait for versioned APIs to implement as part of `tamr-toolbox`.

**Comment by ycharm** [comment]
@OGordon100 is there a process to move these to tbex? @ataghavey code for GRDdelta

**Pull request closed without merging**